### PR TITLE
Decentralized Oracles - Milestone 2 - Graph Api Oracle

### PR DESCRIPTION
**Pull request by GuiGou12358**
Delivery document for Decentralized Oracles - Milestone 2 - Graph Api Oracle

### CODE REVIEW

**Comment by TtomaS7** [comment]
@h4x3rotab will start the review process shortly!

**Review by h4x3rotab — APPROVED** [review]
Excellement submission of the Milestone 2. I'm very happy to see the full demo work smoothly and update the oracle data as expected. Below is the review opinion on each sections.

I only made a change suggestion to the scale coded of the JS oracle implementation, but I think it's good to leave it as a future optimization as the currently impl works as expected.

- Graph API Oracle
	- Very clear code, detailed test cases and documentation
	- Ink! smart contract: graph api consumer
		- No review comment
	- Phat contract: JS Offchain Rollup
		- Prefer to use built-in scale codec for better performance ([your code](https://github.com/decentralized-oracles/graph-api-oracle/blob/895cf8629d8d1d45a65adb98c397f01e74957a47/phat/js/src/index.ts#L15C9-L15C21), [reference docs 1](https://docs.phala.network/developers/bricks-and-blueprints/featured-blueprints/customizing-your-phat-contract#scale-codec), [reference doc 2](https://docs.rs/phat_js/latest/phat_js/#scale-codec))
	- Documentation
		- No review comment
- UI: Smooth experience and the end-to-end test works properly
- SubQuery Indexer: Clear documentation and implementation

File: deliveries/decentralized_oracles-milestone_2.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+|--------|-----------------------------------------|-----------------------------------------------------------------------------------|------------------------------------------------------------------------|
+| 1.     | Phat contract: JS Offchain Rollup       | https://github.com/decentralized-oracles/graph-api-oracle/tree/main/phat/contracts/js_offchain_rollup|                                                                        | 
+| 2.     | Ink! smart contract: graph api consumer | https://github.com/decentralized-oracles/graph-api-oracle/tree/main/ink/contracts/graph_api_consumer|                                                                        | 
+| 3.     | Documentation                           | https://github.com/decentralized-oracles/graph-api-oracle                                |                                                                        | 
+| 4.     | UI                                      | https://github.com/decentralized-oracles/graph-api-ui                            | Demo here: https://graph-api-decentralized-oracle.substrate.fi         |
```
Comment (DoubleOTheven) [inline deliveries/decentralized_oracles-milestone_2.md:48]: There's a lot of unused variables and duplicate import statements. See the error logs in the terminal.

[Here](https://github.com/decentralized-oracles/graph-api-ui/blob/8dc98370647fd003186f2c2bc7d662e28a6362df/src/components/ErrorContract.js#L1) is one

**Comment by TtomaS7** [comment]
Hey team!

Here's the form for payment details: https://forms.gle/BUCKfygkiEvpdYna6.
Once you've filled it out, please let me know and be sure to specify what it pertains to; you can include the link to this PR delivery! Further details can be provided under the section: "Please describe which Milestone you have completed and attach the link to the Milestone delivery."

Should you have any questions, don't hesitate to ask. Thank you so much!

Important Note: PLEASE VERIFY YOUR ADDRESS, WHERE YOU WILL RECEIVE THE FUNDS, BEFORE FILLING OUT THE FORM!
For guidance on how to set up identity, refer to [this guide](https://support.polkadot.network/support/solutions/articles/65000181981-how-to-set-and-clear-an-identity).

File: deliveries/decentralized_oracles-milestone_2.md (hunk 1 above)
Comment (GuiGou12358) [reply deliveries/decentralized_oracles-milestone_2.md:48]: Thanks for spotting these issues. We will fix them.

Comment (arnobase) [reply deliveries/decentralized_oracles-milestone_2.md:48]: > There's a lot of unused variables and duplicate import statements. See the error logs in the terminal.
> 
> [Here](https://github.com/decentralized-oracles/graph-api-ui/blob/8dc98370647fd003186f2c2bc7d662e28a6362df/src/components/ErrorContract.js#L1) is one

thanks for reporting
I've pushed a commit with improvements
https://github.com/decentralized-oracles/graph-api-ui/commit/117bf41adaa9e3e0b19f2978ca108735c8a9cd83

console is indeed much more clean now

**Comment by TtomaS7** [comment]
The payment is sent!
https://polkadot.subsquare.io/treasury/child-bounties/582

You can claim it in 7 day, thank you!

**Pull request merged**